external/java/src/main/java/ch/epfl/dedis/lib/crypto/Bn256Scalar.java

```diff
@@ -0,0 +1,103 @@
+package ch.epfl.dedis.lib.crypto;
+
+import ch.epfl.dedis.lib.crypto.bn256.BN;
+import com.google.protobuf.ByteString;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+
+public class Bn256Scalar implements Scalar {
+    private BigInteger x;
+
+    Bn256Scalar(BigInteger x) {
+        this.x = x;
+    }
+    Bn256Scalar(byte[] buf) {
+        x = new BigInteger(1, buf);
+    }
+    public String toString() {
+        return x.toString();
+    }
+
+    public ByteString toProto() {
+        return ByteString.copyFrom(this.toBytes());
+    }
+
+    public byte[] toBytes() {
+        return bigIntegerToBytes(this.x);
+    }
+
+    public Scalar reduce() {
+        return new Bn256Scalar(this.x.mod(BN.order));
+    }
+
+    public Scalar copy() {
+        return new Bn256Scalar(x);
+    }
+
+    public boolean equals(Scalar other) {
+        return convert(other).x.equals(this.x);
+    }
+
+    public Scalar addOne() {
+        return new Bn256Scalar(this.x.add(BigInteger.ONE));
+    }
+
+    public byte[] getBigEndian() {
+        return this.toBytes();
+    }
+
+    public byte[] getLittleEndian() {
```

Comment:
I tried to look into these but I'm not sure why they exist in the interface. It's something to do with the Ed25519 scalar representation. Perhaps @ineiti knows the history of these methods?

Reply:
Hmm - partial memory reconstruction, might be faulty: there was a problem with having little endian when the scalar was smaller than `2**248`, because then the size of the scalar was not 32 bytes anymore, but only 31 bytes. At least that's what the following tells me: 

https://github.com/dedis/cothority/blob/a2ded818b92275ca725adb2070a87a2c957263e3/external/java/src/main/java/ch/epfl/dedis/lib/crypto/Ed25519Scalar.java#L78

But as it's never used, you can delete it. Just make sure that `getLittleEndian` always returns 32 bytes, even for scalars that are smaller than 2**248.